lib/model/indexhandler.go

```diff
@@ -481,22 +481,14 @@ func (s *indexHandler) logSequenceAnomaly(msg string, extra map[string]any) {
 }
 
 func prepareFileInfoForIndex(f protocol.FileInfo) protocol.FileInfo {
-	// Mark the file as invalid if any of the local bad stuff flags are set.
-	f.RawInvalid = f.IsInvalid()
 	// If the file is marked LocalReceive (i.e., changed locally on a
 	// receive only folder) we do not want it to ever become the
 	// globally best version, invalid or not.
 	if f.IsReceiveOnlyChanged() {
 		f.Version = protocol.Vector{}
 	}
-	// The trailer with the encrypted fileinfo is device local, don't send info
-	// about that to remotes
+	// The trailer with the encrypted fileinfo is device local, announce the size without it to remotes.
 	f.Size -= int64(f.EncryptionTrailerSize)
-	f.EncryptionTrailerSize = 0
-	// never sent externally
-	f.LocalFlags = 0
-	f.VersionHash = nil
-	f.InodeChangeNs = 0
```

Comment:
Explain removal? Also not setting the invalid flag? Is your intention that we now carry local flags on the wire?

Reply:
Ah no, this is a semi-unrelated change: This is redundant since you separated proto types from go types: The conversion method already does the same.
https://github.com/syncthing/syncthing/blob/3d755c87674d2b76d05702eb73daa55e2c2752ce/lib/protocol/bep_fileinfo.go#L142
via:
https://github.com/syncthing/syncthing/blob/main/lib/protocol/bep_index_updates.go#L20

Reply:
As this is a rather large and risky change, lets move this bit into it's own commit: https://github.com/syncthing/syncthing/pull/10173

Reply:
Ah right and about the invalid flag part: As there's now no longer a `RawInvalid` field, the flag is set based on local flags when converting to the wire via `IsInvalid()`: https://github.com/syncthing/syncthing/pull/10170/files/3d755c87674d2b76d05702eb73daa55e2c2752ce#diff-4e823055bae8b0ff914e6ebf37035f58575bcf002a429e8f9121b9be2bdaf446L131

Reply:
Right, right, I saw that but glossed over the significance 👍